Remove go.list and related docs (#4328)
stellar/go: DEVELOPING.md

go get <importpath>@<version>
```

> Before opening a PR make sure to run following command to tidy the module file. It would keep the go.* files current:

Reviewer: ```suggestion
Before opening a PR make sure to run following command to tidy the module file. It will keep the go.* files tidy:
```

go get <importpath>@<version>
```

Go modules files track the minimum dependency required, not the exact dependency version that will be used. To validate the version of the dependency being used update the `go.list` file by running `go mod -m all > go.list`.

Before opening a PR make sure to run these commands to tidy the module files:
```

> go mod tidy

Reviewer: We should keep these three lines, as it is still a good practice to `go mod tidy` after changing dependencies.

When updating or adding dependencies it's critical that we review what the
changes are in those dependencies that we are introducing into our builds. When

> dependencies change the diff for the `go.mod` file may be too complex to
> understand. In those situations use the [golistcmp] tool to get a list of
> changing modules, as well as GitHub links for easy access to diff review.

Reviewer: Could you apply this change?
```suggestion
dependencies change the diff for the `go.mod` file may be complex to
understand. In that situation go dependency by dependency and check
each new or upgraded dependency, and go and check each
dependencies code diffs to see what is being imported.
```
